src/main/java/life/qbic/model/download/QbicDataDownloader.java

```diff
@@ -101,142 +99,75 @@ private static Path getTopDirectory(Path path) {
+              return suffixes.stream().anyMatch(fileName::endsWith);
+            };
+            fileFilter = fileFilter.and(suffixFilter);
           }
-          if (filesDownloadReturnCode != 0) {
+          LOG.info("Initializing download ...");
+          int datasetDownloadReturnCode = downloadDataset(foundDataSets, fileFilter);
+          if (datasetDownloadReturnCode != 0) {
             LOG.error("Error while downloading dataset: " + ident);
-          } else if(!invalidChecksumOccurred) {
-            LOG.info("Download successfully finished");
+          } else if (!invalidChecksumOccurred) {
+            LOG.info("Download successfully finished.");
           }
+        } catch (NullPointerException e) {
+          LOG.error(
+              "Datasets were found by the application server, but could not be found on the datastore server for "
+                  + ident
+                  + "."
+                  + " Try to supply the correct datastore server using a config file!");
         }
+      } else {
+        LOG.info("Nothing to download.");
       }
     }
   }
 
-  /**
-   * Download a given list of data sets
-   *
-   * @param dataSetList A list of data sets
-   * @return 0 if successful, 1 else
-   */
-  private int downloadDataset(List<Map<Sample, List<DataSet>>> dataSetList) {
-    for (Map<Sample, List<DataSet>> dataSetsPerSample : dataSetList) {
-      downloadDataset(dataSetsPerSample);
-    }
-    return 0;
+  private String getFileName(DataSetFile file) {
+    String filePath = file.getPermId().getFilePath();
+    return filePath.substring(filePath.lastIndexOf("/") + 1);
   }
 
-  private void downloadDataset(Map<Sample, List<DataSet>> dataSetsPerSample) {
-
+  private int downloadDataset(Map<Sample, List<DataSet>> dataSetsPerSample, Predicate<DataSetFile> fileFilter) {
+    int returnCode = 0;
     for (Entry<Sample, List<DataSet>> entry : dataSetsPerSample.entrySet()) {
       String sampleCode = entry.getKey().getCode();
       List<DataSet> sampleDatasets = entry.getValue();
       for (DataSet sampleDataset : sampleDatasets) {
         DataSetPermId permID = sampleDataset.getPermId();
-        DataSetFileSearchCriteria criteria = new DataSetFileSearchCriteria();
-        criteria.withDataSet().withCode().thatEquals(permID.getPermId());
-        SearchResult<DataSetFile> result =
-            this.dataStoreServer.searchFiles(sessionToken, criteria,
-                new DataSetFileFetchOptions());
-        List<DataSetFile> filteredDataSetFiles = QbicDataFinder.withoutDirectories(result.getObjects());
-        final DownloadRequest downloadRequest = new DownloadRequest(filteredDataSetFiles,
-            sampleCode, DEFAULT_DOWNLOAD_ATTEMPTS);
-        downloadFiles(downloadRequest);
+        List<DataSetFile> files = qbicDataFinder.getFiles(permID, fileFilter);
+        if (files.isEmpty()) {
+          LOG.info("Nothing to download for dataset " + sampleDataset.getPermId() + ".");
```

Comment:
Nit: Is this relevant information concerning suffix filtering? 
Dependent on the suffix this can spam the log if a project has a multitude of datasets 

`15:07:02.631 [main] INFO  life.qbic.model.download.QbicDataDownloader - Nothing to download for dataset 20220923142207884-971611.`